pkg/exporter/syslogexporter/exporter.go

```diff
@@ -121,32 +125,47 @@ func (se *syslogexporter) getTimestamp(record plog.LogRecord) time.Time {
 }
 
 func (se *syslogexporter) pushLogsData(ctx context.Context, ld plog.Logs) error {
-	s, err := Connect(se.logger, se.config, se.tlsConfig, se.hostname, se.pid, se.app)
-	if err != nil {
-		return fmt.Errorf("error connecting to syslog server: %s", err)
+	var errs []error
+	s, errConn := Connect(se.logger, se.config, se.tlsConfig, se.hostname, se.pid, se.app)
+	if errConn != nil {
+		errs = append(errs, errConn)
+		errStr := fmt.Sprintf("Unable to send syslog messages: %s", strings.Join(errorListToStringSlice(errs), ", "))
+		return errors.New(errStr)
 	}
```

Comment:
We should return `consumererror` which contains unsent logs. Please see the following exmaples in sumoplogicexporter:

- https://github.com/SumoLogic/sumologic-otel-collector/blob/dc1c3185755b21b2252a8f7d01a5a596e1134614/pkg/exporter/sumologicexporter/exporter.go#L180-L182
- https://github.com/SumoLogic/sumologic-otel-collector/blob/dc1c3185755b21b2252a8f7d01a5a596e1134614/pkg/exporter/sumologicexporter/exporter.go#L250-L253

Reply:
but in syslog exporter log messages may contain token which is required by sumologic ;/

Reply:
Good point, but I don't think that returned logs are printed anywhere

Reply:
so what should we do here. This was a deliberate choice after discussing with Kasia. This commit focuses more on error handling.

Reply:
it should return: `return consumererror.NewLogs(multierr.Combine(errs...), ld)`

where `ld` contains unsent logs

Reply:
and errs is a list of errors to print for customer

Reply:
pushed changes to support `consumererror` with dropped logs

Reply:
@rnishtala-sumo Could you check if returned logs are printed, see: https://github.com/SumoLogic/sumologic-otel-collector/pull/986#discussion_r1108339762?

Reply:
it doesn't look like it
```2023-02-16T11:29:33.070-0500	error	exporterhelper/queued_retry.go:303	Exporting failed. Dropping data. Try enabling sending_queue to survive temporary failures.	{"kind": "exporter", "data_type": "logs", "name": "syslog", "dropped_items": 1}```